doc/decisions/DR014-Management-policy-responses.md

```diff
@@ -0,0 +1,307 @@
+    ManagedTrait.imbueTo(policy)
+```
+
+Such that it opts-in to managing all entities with the file trait in
+their set, aside from caches:
+
+```python
+policies = [
+    { 'managed': {...} },  # Main document
+    { 'managed': {...} },  # Generated images
+    { 'managed': {...} },  # Generated maps
+    {},                    # Temporary cache
+]
+```
+
+Any other queries for trait sets without the file trait would be empty
+`{}` indicating un-managed status.
+
+#### Pros
+
+- Hosts can query specific and precise trait sets, but the manager does
+  not need to worry about all possible permutations.
+
+#### Cons
+
+- There is no way for a host to determine what traits can be resolved
+  (or persisted) for any given entity, and so must tolerate missing data
+  at a later date (e.g. the color space for the `colorManaged` trait, if
+  the manager can't ever provide this).
+
+### Option 2 - Exact Matching
+
+The manager should respond only to sets where it is capable of resolving
+or persisting all of the requested trait set:
+
+```python
+if traitSet == { FileTrait.kId }:
+    ManagedTrait.imbueTo(policy)
+```
+
+Such that it _only_ opts into managing the specifically limited trait
+set that it can support:
+
+```python
+policies = [
+    {},  # Main document
+    {},  # Generated Images
+    {},  # Generated maps
+    {}   # Temporary caches
+]
+```
+
+The Host would receive empty responses (`{}`) for all of the original
+trait sets, and so must potentially decompose and re-try its queried
+trait sets to reveal some supported sub-set. For example, if ultimately
+it just needed the file path, it could try these combinations:
+
+```python
+traitSets = [
+    { 'file', 'nodeGraph' },
+    { 'file', 'image', 'colorManaged' },
+    { 'file', 'image' },
+    { 'file', 'image', 'latLong', 'colorManaged' },
+    { 'file', 'image', 'latLong' },
+    { 'file', 'cache' },
+    { 'file' }
+]
+```
+
+Resulting in:
+
+```python
+policies = [
+    {},
+    {},
+    {},
+    {},
+    {},
+    {},
+    { 'managed': {...} }
+]
+```
+
+#### Pros
+
+- Conveys only what traits are supported to the host, allowing
+  behaviour to be properly adapted to the manager's capabilities.
+
+#### Cons
+
+- No way to know that actually the manager was interested in managing
+  all but one of the requested trait sets.
+- Host implementation is significantly more complicated.
+- No way to hint at any additional traits that may be resolvable
+  outside the typing trait set.
+
+### Option 3 - Supported Traits Response
+
+The host should also include any additional traits that it may attempt
+to resolve in the future, e.g. if it supported the ability to customise
+the file format used for writing images, or cache lifetime by resolving
+a custom trait:
+
+```python
+traitSets =[
+    { 'file', 'nodeGraph' },
+    { 'file', 'image', 'colorManaged', 'fileFormatOptions' }
+    { 'file', 'image', 'latLong', 'colorManaged', 'fileFormatOptions' }
+    { 'file', 'cache', 'retention' }
```

Comment:
[question] In this example `'cache'` causes a short-circuit regardless what other traits are requested.

What if the manager could persist data for other kinds of cache e.g. `{ 'dbquery', 'cache' }`?  Then we have the situation where the combination of traits must be taken as a whole in order to determine whether the trait set is `'managed'`. I.e. the meaning of one trait is dependent on presence or absence of another.

I think we've discussed this can of worms before.

Might be worth clarifying in the intro that traits in a trait set don't strongly interact with each other and should be considered in isolation?

Reply:
I attempted to remove some of the ambiguity in the [rewrite](From that perspective, a response containing a bunch of traits we didn't ask for seems odd.), but it might need more, let me know what you think.